nixos/modules/services/networking/ejabberd.nix

```diff
@@ -12,6 +12,7 @@ let
   '';
 
   settingsFormat = pkgs.formats.yaml {};
+  configFile = settingsFormat.generate "config.yaml" cfg.settings;
 
   ectl = ''${cfg.package}/bin/ejabberdctl --config "${cfg.configFile}" --ctl-config "${ctlcfg}" --spool "${cfg.spoolDir}" --logs "${cfg.logsDir}"'';
```

Comment:
Just have to remember to replace `cfg.configFile` with `configFile` when the former is removed, otherwise LGTM